package org.rmj.api.test.desktop;

import java.io.IOException;
import java.util.Calendar;
import java.util.HashMap;
import java.util.Map;
import org.json.simple.JSONObject;
import org.json.simple.parser.JSONParser;
import org.json.simple.parser.ParseException;
import org.rmj.api.test.login.WebClient;
import org.rmj.appdriver.MySQLAESCrypt;
import org.rmj.appdriver.SQLUtil;

public class testRequestEmployee {
    public static void main(String[] args) {
        String sURL = "https://restgk.guanzongroup.com.ph/lospedritos/accounts/getEmployee.php";
        
        Map<String, String> headers = getHeader();
        
        JSONObject param = new JSONObject();
        //param.put("employno", "00100538711");             //option 1
        //param.put("employno", "001-005387-11");           //option 2
        param.put("employno", "M00111005387");     //option 3
                
        String response;
        try {
            response = WebClient.sendHTTP(sURL, param.toJSONString(), (HashMap<String, String>) headers);
            if(response == null){
                System.err.println("No Response"); //return value
                System.exit(1);
            } 
            
            JSONParser loParser = new JSONParser();
            JSONObject loJSON = (JSONObject) loParser.parse(response);
            
            if (!"success".equals((String) loJSON.get("result"))){
                loJSON = (JSONObject) loParser.parse(loJSON.get("error").toString());
                
                response = (String) loJSON.get("message");
                
                System.err.println(response); //return value
                System.exit(1);
            } else {
                response = MySQLAESCrypt.Decrypt((String) loJSON.get("payload"), "20190625");
            
                System.out.println(response); //return value
                System.exit(0);
            }
        } catch (IOException | ParseException ex) {
            System.err.println(ex.getMessage());
            System.exit(1);
        }
    }
    
    private static HashMap getHeader(){
        String clientid = "GGC_BM001";
        String productid = "gRider";
        String imei = "GMC_SEG09";
        String user = "M001111122";
        String log = "";
        
        Calendar calendar = Calendar.getInstance();
        Map<String, String> headers = 
                        new HashMap<String, String>();
        headers.put("Accept", "application/json");
        headers.put("Content-Type", "application/json");
        headers.put("g-api-id", productid);
        headers.put("g-api-imei", imei);
        
        headers.put("g-api-key", SQLUtil.dateFormat(calendar.getTime(), "yyyyMMddHHmmss"));        
        headers.put("g-api-hash", org.apache.commons.codec.digest.DigestUtils.md5Hex((String)headers.get("g-api-imei") + (String)headers.get("g-api-key")));
        headers.put("g-api-client", clientid);    
        headers.put("g-api-user", user);    
        headers.put("g-api-log", log);    
        headers.put("g-char-request", "UTF-8");
        headers.put("g-api-token", "");    
        
        return (HashMap) headers;
    }
}
